volume/local/local_unix.go

```diff
@@ -143,6 +143,11 @@ func (v *localVolume) mount() error {
 		}
 	}
 	err := mount.Mount(v.opts.MountDevice, v.path, v.opts.MountType, mountOpts)
+	if err != nil {
+		if password := getPassword(v.opts.MountOpts); password != "" {
+			err = errors.New(strings.Replace(err.Error(), "password="+password, "password=********", 1))
+		}
```

Comment:
The only concern I have with this, is that this can hide the underlying error-type, which would prevent detecting (e.g.) `errors.Is(err, fs.NotExist)`.

It looks like currently we don't do any detection of error-types, and consider any error to be a "500" (internal server error) (probably not correct either for this case 🤔 but that's a separate issue); https://github.com/moby/moby/blob/4cac624fefefb7bb6640f205587d0dc0a9651b27/volume/local/local.go#L315-L317

Let me at least get some eyes from @cpuguy83 on this as well